### DISCUSSION TITLE
Max topics for subscribeToTopics/unsubscribeFromTopics methods

### DISCUSSION THREAD

**Discussion by kccodeguy** [kreait/firebase-php#605] [Q&A]
I see in the documentation that the subscribeToTopics/unsubscribeFromTopics methods can have 1000 tokens max but there is no mention of the max number of topics. Is there a limit on the number of topics for these calls. Thanks.

**Comment by jeromegamez** [comment]
The SDK will create one individual request for each topic that you pass to the method and send all requests asynchronously. So, there _should_™ be no limit on the number of topics. I've never tried to subscribe a device to 1000 topics, so I can't tell you what will happen, but it _could_ be that the requests get rate limited (but again, I don't know, so this is just a guess 😅)
